--- a/crates/muvm/src/guest/x11.rs
+++ b/crates/muvm/src/guest/x11.rs
@@ -16,7 +16,8 @@
     if !host_display.starts_with(':') {
         return Err(anyhow!("Invalid host DISPLAY"));
     }
-    let host_display = &host_display[1..];
+    // Xauthority stores the display number, without an optional screen suffix.
+    let host_display = host_display[1..].split('.').next().unwrap();
 
     // SAFETY: Safe if and only if `muvm-guest` program is not multithreaded.
     // See https://doc.rust-lang.org/std/env/fn.set_var.html#safety
